packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/summary.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { shape, string, func } from 'prop-types';
+import { Edit2 as EditIcon } from 'react-feather';
+import { useSummary } from '@magento/peregrine/lib/talons/CheckoutPage/PaymentInformation/useSummary';
+
+import Icon from '../../Icon';
+import { mergeClasses } from '../../../classify';
+
+import summaryGQLOperations from './summary.gql';
+
+import defaultClasses from './summary.css';
+
+const Summary = props => {
+    const { classes: propClasses, onEdit, paymentNonce } = props;
+
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    const talonProps = useSummary({ operations: summaryGQLOperations });
+
+    const { billingAddress, isBillingAddressSame } = talonProps;
+
+    const billingAddressSummary =
+        !isBillingAddressSame && billingAddress ? (
+            /**
+             * Added ID for tests
+             */
+            <div
+                id={'addressSummary'}
+                className={classes.address_summary_container}
+            >
+                <div>
+                    <span className={classes.first_name}>
+                        {billingAddress.firstName}
+                    </span>
+                    <span className={classes.last_name}>
```

Comment:
I think this could be simplified. Check out [how Tommy did it](https://github.com/magento/pwa-studio/pull/2285/files#diff-746541a4b0f7d43762fed892f9cec1fa). You just need one parent container and then some child spans that include template strings:

```js
<div className={classes.addressSummary>
  <span className={classes.name}{`${firstname lastname}`}</span>
  ...
  ...
</div>
```

Reply:
I had that but purposefully changed it because @jimbo mentioned we have get as granular as we can with `classNames` so end users can have full control over the styling. That is the reason why I broke them up into individual spans.

Reply:
Ok fair point - I will say that nowhere else in the app do we split presentation of firstname and lastname into separate spans. That said, I can see the confusion in the guidelines you were given :)

Reply:
I'm fine with how @revanth0212 has done it here. 👍